src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -231,18 +240,36 @@ public void sendAsync(RowMap r, AbstractAsyncProducer.CallbackCompleter cc) thro
 
 		KafkaCallback callback = new KafkaCallback(cc, r.getNextPosition(), r.getRowIdentity(), value,
 				this.succeededMessageCount, this.failedMessageCount, this.succeededMessageMeter, this.failedMessageMeter,
-				this.deadLetterTopic, this.context, this);
+				this.topic, this.deadLetterTopic, this.context, this);
 
 		sendAsync(record, callback);
 	}
 
-	public void sendFallbackAsync(String topic, RowIdentity fallbackRecord, KafkaCallback callback, RecordMetadata md, Exception reason) {
-		try {
-			ProducerRecord<String, String> record = makeFallbackRecord(topic, fallbackRecord, reason);
-			sendAsync(record, callback);
-		} catch (Exception fallbackEx) {
-			callback.onCompletion(md, fallbackEx);
+	private ExecutorService getBackgroundExecutor() {
+		if (this.backgroundExecutor == null) {
+			this.backgroundExecutor = Executors.newSingleThreadExecutor();
```

Comment:
If possible it's good to give named thread factories to help with log debugging.  Not sure we have a named thread factory easily available from any of the dependencies.

Reply:
I looked around for thread factories in scope (I like to `setDaemon(false)`), but there was nothing readily available, and I think the logs should make it reasonably clear.